src/routes/example/[slug]/+page.svelte

```diff
@@ -40,11 +16,9 @@
 	hljs.registerLanguage('svelte', hljsDefineSvelte);
 	hljsDefineSvelte(hljs);
 
-	export let slug;
-	export let data;
-
-
-	export let active = '+page';
+	export let data
+	let {slug, content, active} = data
+	$: ({slug, content, active} = data)
```

Comment:
Why not just have whats on line 21? Why declare these variables on line 20 first?

Reply:
It's taken from the "lazy" pattern from the migration guide: https://github.com/sveltejs/kit/discussions/5774#discussioncomment-3292707

Not sure if it's possible without the other lines. One could certainly use `data.slug` etc. throughout the rest of the code.

```diff
<script>
-  /** @type {string} */
-  export let foo;
-
-  /** @type {boolean} */
-  export let bar;
-
-  $: console.log({ foo, bar });
+  /** @type {import('./$types').PageData */
+  export let data;
+  let { foo, bar } = data;
+  $: ({ foo, bar } = data); // so it stays in sync when `data` changes
+
+  $: console.log({ foo, bar });
</script>
```

Reply:
Here's the explanation: https://twitter.com/geoffrich_/status/1564297339946553346

Reply:
That's very helpful!